### PR TITLE
feat: rechunk support

### PR DESCRIPTION
**Pull request by xynydev**
(also some autoformatting)

### CODE REVIEW

**Comment by xynydev** [comment]
I'm unable to get this working, but not sure what the issue is. Maybe someone else can figure this out?
https://github.com/xynydev/linuXYZ/actions/runs/12124482608/job/33802828924

**Comment by fiftydinar** [comment]
Ideally, `sudo` should only be running when `rechunk` is executed, otherwise non-root execution

**Comment by xynydev** [comment]
In principle I agree, but I'm not sure what the actual security drawbacks would be on GitHub Actions. Getting this working would be my first priority.

**Comment by fiftydinar** [comment]
https://rhysd.github.io/actionlint/ doesn't report anything useful

**Comment by fiftydinar** [comment]
Maybe you need to append `BUILD_OPTS` array, rather than declare it as new?

**Comment by gmpinder** [comment]
Ok I think I got this working https://github.com/gmpinder/testos/actions/runs/12131970193/job/33825026568

**Comment by fiftydinar** [comment]
> Ok I think I got this working https://github.com/gmpinder/testos/actions/runs/12131970193/job/33825026568

I confirm it works.

Version is changed from `yyyymmdd` format to `yymmdd`, that's Rechunk behavior, so it works.

I approve this.

**Review by fiftydinar — DISMISSED** [review]

File: action.yml [hunk 1]
```diff
@@ -51,47 +51,56 @@ inputs:
...
+      Example: `ublue-os`
     required: false
     default: ${{ github.repository_owner }}
+  rechunk:
+    description: |
+      Rechunk the ostree-based result images with [github.com/hhd-dev/rechunk](https://github.com/hhd-dev/rechunk) for more efficient diffs and updates. (lower image size, better download speed, better update resuming)
+
+      Will make your builds considerably slower. This is an experimental option, as it can cause issues with file permissions in some scenarios, so enable on your own risk.
+
+      Uses `--squash` build opt to further reduce the image size.
```
Comment (xynydev) [inline action.yml:68]: I can't see enabling the `rechunk` automatically enabling squash anywhere here? Is this something done in the CLI, or perhaps a typo (`uses` vs. `use`)?

**Comment by xynydev** [comment]
I approve this too, but I can't because I started the PR, so one of you should do a final look-over an approve.

**Review by fiftydinar — APPROVED** [review]

**Pull request merged**